internal/service/alarms/internal/resourceserver/resourceserver.go

```diff
@@ -0,0 +1,65 @@
+package resourceserver
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"sync"
+
+	"github.com/openshift-kni/oran-o2ims/internal/service/common/clients"
+	"github.com/openshift-kni/oran-o2ims/internal/service/common/clients/resources"
+)
+
+var (
+	// url is the URL of the resource server
+	url  string
+	once sync.Once
+)
+
+// GetResourceTypes gets the list of resource types from the resource server
+func GetResourceTypes(ctx context.Context) (*[]resources.ResourceType, error) {
+	c, err := getClient()
+	if err != nil {
+		return nil, err
+	}
+
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, clients.SingleRequestTimeout)
+	defer cancel()
+
+	resp, err := c.GetResourceTypesWithResponse(ctxWithTimeout, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute Get operation: %w", err)
+	}
+
+	if resp.StatusCode() != http.StatusOK {
+		return nil, fmt.Errorf("status code different from 200: %d", resp.StatusCode())
+	}
+
+	return resp.JSON200, nil
+}
+
+// getClient gets the client for the resource server
+func getClient() (*resources.ClientWithResponses, error) {
+	var err error
+	once.Do(func() {
+		// TODO: update the URL to the correct one once the service is deployed
+		url = os.Getenv("RESOURCE_SERVER_URL")
```

Comment:
The service is already deployed.  It's not using the new API, but theoretically it should still work.  

See `GetServiceURL` in https://github.com/openshift-kni/oran-o2ims/pull/351 to build the URL.

Reply:
Thanks. Good to know. I created this pr without taking into account the actual provisioning. Will update it

Reply:
updated